imageprocessor/src/main/java/pl/edu/uj/kimage/processingFlow/FlowStep.java

```diff
@@ -0,0 +1,42 @@
+package pl.edu.uj.kimage.processingFlow;
+
+import pl.edu.uj.kimage.Plugin;
+import pl.edu.uj.kimage.eventbus.Event;
+
+public class FlowStep {
+    private final int stepId;
+    private final Plugin stepRelatedPlugin;
+
+    public FlowStep(int stepId, Plugin stepRelatedPlugin) {
+        this.stepId = stepId;
+        this.stepRelatedPlugin = stepRelatedPlugin;
+    }
+
+    public int getStepId() {
+        return stepId;
+    }
+
+    public void processEvent(Event event)
+    {
+        stepRelatedPlugin.processEvent(event);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        FlowStep flowStep = (FlowStep) o;
+
+        if (stepId != flowStep.stepId) return false;
+        return stepRelatedPlugin != null ? stepRelatedPlugin.getClass().equals(flowStep.stepRelatedPlugin.getClass()) : flowStep.stepRelatedPlugin == null;
```

Comment:
Yes I know, but you should be very careful in choosing classes where you override those methods. If you choose wrong it could cause errors